src/Dodo.HttpClient.ResiliencePolicies/HttpClientSettings.cs

```diff
@@ -27,17 +29,24 @@ public class HttpClientSettings
 			TimeSpan.FromMilliseconds(Defaults.Timeout.HttpClientTimeoutInMilliseconds),
 			TimeSpan.FromMilliseconds(Defaults.Timeout.TimeoutPerTryInMilliseconds),
 			retrySettings,
-			circuitBreakerSettings)
+			circuitBreakerSettings,
+			TimeSpan.FromMilliseconds(Defaults.Timeout.TimeoutOverallInMillicesons))
 		{
 		}
 
 		public HttpClientSettings(
 			TimeSpan httpClientTimeout,
 			TimeSpan timeoutPerTry,
 			IRetrySettings retrySettings,
-			ICircuitBreakerSettings circuitBreakerSettings)
+			ICircuitBreakerSettings circuitBreakerSettings,
+			TimeSpan? timeoutOverall = null)
 		{
-			HttpClientTimeout = httpClientTimeout;
+			TimeoutOverall = timeoutOverall;
```

Comment:
It is a bit tricky thing, but it is possible to do. Actually HttpClientSettings already have information about selected retry strategy. So, it can calculate the sum of all retry attempts. It will be not exact value for strategy with jitter backoffs, but good enough to validate overall timeout.